lib/commands/touch.js

```
@@ -0,0 +1,274 @@
+import log from '../logger';
+import _ from 'lodash';
+import androidHelpers from '../android-helpers';
+import B from 'bluebird';
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.doTouchAction = async function (action, opts) {
+  switch (action) {
+    case 'tap':
+      return await this.tap(opts.element, opts.x, opts.y, opts.count);
+    case 'press':
+      return await this.touchDown(opts.element, opts.x, opts.y);
+    case 'release':
+      return await this.touchUp(opts.element, opts.x, opts.y);
+    case 'moveTo':
+      return await this.touchMove(opts.element, opts.x, opts.y);
+    case 'wait':
+      return await B.delay(opts.ms);
+    case 'longPress':
+      if (typeof opts.duration === 'undefined' || !opts.duration) {
+        opts.duration = 1000;
+      }
+      return await this.touchLongClick(opts.element, opts.x, opts.y, opts.duration);
+    case 'cancel':
+      // TODO: clarify behavior of 'cancel' action and fix this
+      log.warn("Cancel action currently has no effect");
+      break;
+    default:
+      return `unknown action ${action}`;
```

why would we return a string? do we not want to throw an error?